wicker/core/datasets.py

```diff
@@ -303,3 +455,76 @@ def dataset_size(self) -> int:
             int: total dataset size in bytes
         """
         return self._get_dataset_partition_size()
+
+
+def get_filesystem_config(config_name: str, config: WickerConfig) -> Optional[WickerFileSystemConfig]:
+    for filesystem_config in config.filesystem_configs:
+        if filesystem_config.config_name == config_name:
+            return filesystem_config
+    return None
+
+
+def build_dataset(
```

Comment:
👍 . Minor thing lean to you is from the testable/readable perspective, I would think make 2 private sub function like _build_s3_dataset _build_filesystem_dataset. 
Anyway, much better than the original S3Dataset interface. Thanks